src/Testing/testCiphers.cpp

```diff
@@ -0,0 +1,62 @@
+//! Unit Tests for MPAGSCipher Cipher Classes
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include "Cipher.hpp"
+#include "CipherFactory.hpp"
+#include "ProcessCommandLine.hpp"
+#include "CaesarCipher.hpp"
+#include "PlayfairCipher.hpp"
+#include "ViginereCipher.hpp"
+
+#include <memory> 
+#include <string>
+#include <vector>
+
+bool testCipher(const std::unique_ptr<Cipher>& cipher, const CipherMode mode,
+                const std::string& inputText, const std::string& outputText) {
```

Comment:
Forgot to say before that the cipher should be passed as a simple reference:
```suggestion
bool testCipher(const Cipher& cipher, const CipherMode mode,
                const std::string& inputText, const std::string& outputText) {
```
Otherwise your function is limited to acting on ciphers whose lifetime is managed by unique_ptr's.  Whereas with the above change it can act on any cipher.